File: 준영/LV2/튜플.py
def solution(s):
    answer = []
    arr = s.split('},')
    for i in range(len(arr)):
        arr[i] = arr[i].strip('{')
        arr[i] = arr[i].strip('}')
        arr[i] = list(map(int, arr[i].split(',')))

    arr.sort(key=lambda x: len(x))

    cnt_arr = [0] * 100001
    
    for i in arr:
        for j in i:
            if cnt_arr[j] == 0:
                cnt_arr[j] = 1
                answer.append(j)

    return answer

s = "{{2},{2,1},{2,1,3},{2,1,3,4}}"
solution(s)

File: 준영/프로그래머스/LV2/튜플.js
function solution(s) {
  var answer = []
  var arr = s.split('},')
  for (let i = 0; i < arr.length; i++) {
    arr[i] = arr[i].replace(/{+/, '')
    arr[i] = arr[i].replace(/}+/, '')
    arr[i] = arr[i].split(',').map(Number)
  }

  arr.sort((a, b) => a.length - b.length)

  var cnt_arr = Array(100001).fill(0)

  for (const i of arr) {
    for (const j of i) {
      if (cnt_arr[j] === 0) {
        cnt_arr[j] = 1
        answer.push(j)
      }
    }
  }
  return answer
}